x/pocketcore/keeper/service_test.go

```diff
@@ -66,25 +66,60 @@ func TestKeeper_HandleRelay(t *testing.T) {
 		t.Fatalf(er.Error())
 	}
 	validRelay.Proof.Signature = hex.EncodeToString(clientSig)
-	defer gock.Off() // Flush pending mocks after test execution
-
-	gock.New("https://www.google.com:443").
-		Post("/").
-		Reply(200).
-		BodyString("bar")
 
 	mockCtx.On("KVStore", keeper.storeKey).Return(ctx.KVStore(keeper.storeKey))
 	mockCtx.On("KVStore", keys["pos"]).Return(ctx.KVStore(keys["pos"]))
 	mockCtx.On("KVStore", keys["params"]).Return(ctx.KVStore(keys["params"]))
 	mockCtx.On("KVStore", keys["application"]).Return(ctx.KVStore(keys["application"]))
-	mockCtx.On("BlockHeight").Return(ctx.BlockHeight())
+	mockCtx.On("BlockHeight").Return(currentBlockHeight)
 	mockCtx.On("PrevCtx", int64(976)).Return(ctx, nil)
 	mockCtx.On("PrevCtx", keeper.GetLatestSessionBlockHeight(mockCtx)).Return(ctx, nil)
 	mockCtx.On("Logger").Return(ctx.Logger())
+	return
+}
+
+func TestKeeper_HandleRelay(t *testing.T) {
+	defer gock.Off() // Flush pending mocks after test execution
 
+	gock.New("https://www.google.com:443").
```

Comment:
For my personal learning, can you explain why we need this?

Reply:
To be honest, it was mostly copy and pasta, but I do have experience in mocking out requests in tests. We don't want to send a real request, certainly not to google, and so we mock the request out with an expected request which allows us to test the intended function.